File: src/main/java/com/clearcont/clearcontapp/helpers/DownloadExcel.java
package com.clearcont.clearcontapp.helpers;

import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.Anchor;
import com.vaadin.flow.component.icon.Icon;
import com.vaadin.flow.server.StreamResource;
import org.jetbrains.annotations.NotNull;

public abstract class DownloadExcel {
    @NotNull
    public static Anchor generateExcelDownloadLink(StreamResource excelStreamResource) {
        Anchor downloadLink = new Anchor(excelStreamResource, "");
        downloadLink.getElement().setAttribute("download", true);
        Button exportButton = new Button("Exportar para Excel");
        exportButton.getStyle().setBackground("darkgreen");
        downloadLink.add(exportButton);
        exportButton.setIcon(new Icon("download"));
        exportButton.getElement().setAttribute("download", true);
        exportButton.getElement().setAttribute("href", excelStreamResource);
        return downloadLink;
    }

}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/helpers/DownloadExcel.kt
package br.com.clearcont.clearcontwebapp.helpers

import com.vaadin.flow.component.button.Button
import com.vaadin.flow.component.html.Anchor
import com.vaadin.flow.component.icon.Icon
import com.vaadin.flow.server.StreamResource

fun generateExcelDownloadLink(excelStreamResource: StreamResource?): Anchor {
    val exportButton = Button("Exportar para Excel").apply {
        style.setBackground("darkgreen")
        icon = Icon("download")
        element.setAttribute("download", true)
        element.setAttribute("href", excelStreamResource)
    }
    val downloadLink = Anchor(excelStreamResource, "").apply {
        element.setAttribute("download", true)
        add(exportButton)
    }
    return downloadLink
}
